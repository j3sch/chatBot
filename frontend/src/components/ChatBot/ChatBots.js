import { useSelector, useDispatch } from 'react-redux';

import { ChangeChatId } from '../../store/actions/Chatbot';

function Index({ name, img, text, time, event, id }) {
	const dispatch = useDispatch();

	const currentChatId = useSelector((state) => state.chatState.ChatId);

	const changeChatId = (number) => {
		dispatch(ChangeChatId(number));
	};

	return (
		<div
			data-testid="id1"
			id={'chatBot' + id}
			className={
				'pl-4 pr-4 cursor-pointer ' +
				(currentChatId === id
					? 'bg-indigo-100 dark:bg-slate-500'
					: 'bg-white hover:bg-indigo-50 dark:bg-slate-700 dark:hover:bg-slate-600')
			}
			onClick={() => {
				changeChatId(id);
				if (event != undefined) {
					event();
				}
			}}
		>
			<div className="pt-4">
				<div>
					<div className="flex flex-row justify-between ">
						<div className="flex flex-row items-center">
							<img src={img} alt="Business Mann 42" className="h-16 w-16 rounded-full" />
							<div className="ml-4 mr-4 flex-1">
								<h1 className="text-xl font-bold dark:text-white">{name}</h1>
								<h1 className="text max-h-12 overflow-hidden dark:text-white">{text}</h1>
							</div>
						</div>
						<div className="flex flex-row items-center">
							<h1 className="text">{time}</h1>
						</div>
					</div>
				</div>
				<div className="ml-20 mt-4 -mr-4 h-px bg-slate-300 dark:bg-slate-400" />
			</div>
		</div>
	);
}

export default Index;
