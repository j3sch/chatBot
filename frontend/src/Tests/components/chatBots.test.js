import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ChatBots from '../../components/ChatBot/ChatBots';

describe('ChatBots', () => {
	let store;

	beforeEach(() => {
		const reducer = (state, action) => {
			switch (action.type) {
				case 'ChangeChatId':
					return { ...state, ChatId: action.payload };
				case 'ClearChat':
					return { ...state, chat: [] };
				default:
					return state;
			}
		};
		store = createStore(reducer, { chatState: { ChatId: 1 } });
	});

	it('ChatBots Test', () => {
		const mockedEvent = jest.fn();

		render(
			<Provider store={store}>
				<ChatBots
					name="Test Name"
					img="test-image.jpg"
					text="Test Text"
					time="Test Time"
					event={mockedEvent}
					id={2}
				/>
			</Provider>,
		);

		const component = screen.getByTestId('id1');

		expect(component).toBeInTheDocument();
		expect(component).toHaveTextContent('Test Name');
		expect(component).toHaveTextContent('Test Text');
		expect(component).toHaveTextContent('Test Time');
	});

	test('matches snapshot', () => {
		const mockedEvent = jest.fn();

		const tree = renderer
			.create(
				<Provider store={store}>
					<ChatBots
						name="Test Name"
						img="test-image.jpg"
						text="Test Text"
						time="Test Time"
						event={mockedEvent}
						id={2}
					/>
				</Provider>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
