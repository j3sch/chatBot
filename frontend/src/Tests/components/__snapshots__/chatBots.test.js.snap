// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBots matches snapshot 1`] = `
<div
  className="pl-4 pr-4 cursor-pointer bg-white hover:bg-indigo-50 dark:bg-slate-700 dark:hover:bg-slate-600"
  data-testid="id1"
  id="chatBot2"
  onClick={[Function]}
>
  <div
    className="pt-4"
  >
    <div>
      <div
        className="flex flex-row justify-between "
      >
        <div
          className="flex flex-row items-center"
        >
          <img
            alt="Business Mann 42"
            className="h-16 w-16 rounded-full"
            src="test-image.jpg"
          />
          <div
            className="ml-4 mr-4 flex-1"
          >
            <h1
              className="text-xl font-bold dark:text-white"
            >
              Test Name
            </h1>
            <h1
              className="text max-h-12 overflow-hidden dark:text-white"
            >
              Test Text
            </h1>
          </div>
        </div>
        <div
          className="flex flex-row items-center"
        >
          <h1
            className="text"
          >
            Test Time
          </h1>
        </div>
      </div>
    </div>
    <div
      className="ml-20 mt-4 -mr-4 h-px bg-slate-300 dark:bg-slate-400"
    />
  </div>
</div>
`;
